### PR TITLE
Add "count" method

### PR DESCRIPTION
**Pull request by binkpitch**
Add "count" method as described in https://github.com/supercharge/set/issues/1.

### CODE REVIEW

File: src/set.ts [hunk 1]
```diff
@@ -217,4 +217,23 @@ export class SuperchargedSet<T> implements Iterable<T> {
...
+   * Returns the number of items matching the given predicate.
+   *
+   * @param {Function} predicate
+   *
+   * @returns {number}
+   */
+  count (predicate: (item: T) => boolean): number {
+    let count = 0
+
+    for (const value of this.set.values()) {
```
Comment (marcuspoehls) [inline src/set.ts:231]: The set already has the `.filter(predicate)` and `.size()` methods. I would like to see the `.count(predicate)` using both. What do you think?

**Review by marcuspoehls — COMMENTED** [review]
Hey, I left a comment in the code related to using the existing `.filter()` and `.size` methods for the new `.count()` method. This would reuse existing methods without doing all the looping yourself. What do you think?

File: src/set.ts (hunk 1 above)
Comment (binkpitch) [reply src/set.ts]: I think your approach of reusing methods is better. I have submit a new refractor commit.

**Comment by marcuspoehls** [comment]
@binkpitch Looks good, thank you!

**Pull request merged**